core/modules/locale/locale.module

```diff
@@ -563,12 +563,21 @@ function locale($string = NULL, $context = NULL, $langcode = NULL) {
         // Refresh database stored cache of translations for given language.
         // We only store short strings used in current version, to improve
         // performance and consume less memory.
-        $result = db_query("SELECT s.source, s.context, t.translation, t.language FROM {locales_source} s LEFT JOIN {locales_target} t ON s.lid = t.lid AND t.language = :language WHERE s.version = :version AND LENGTH(s.source) < :length", array(':language' => $langcode, ':version' => BACKDROP_VERSION, ':length' => config_get('locale.settings', 'cache_length')));
-        foreach ($result as $data) {
-          $locale_t[$langcode][$data->context][$data->source] = (empty($data->translation) ? TRUE : $data->translation);
+        try {
+          $result = db_query("SELECT s.source, s.context, t.translation, t.language FROM {locales_source} s LEFT JOIN {locales_target} t ON s.lid = t.lid AND t.language = :language WHERE s.version = :version AND LENGTH(s.source) < :length", array(
+            ':language' => $langcode,
+            ':version' => BACKDROP_VERSION,
+            ':length' => config_get('locale.settings', 'cache_length'),
+          ));
+          foreach ($result as $data) {
+            $locale_t[$langcode][$data->context][$data->source] = (empty($data->translation) ? TRUE : $data->translation);
+          }
+          cache()->set('locale:' . $langcode, $locale_t[$langcode]);
+          lock_release('locale_cache_' . $langcode);
+        }
+        catch (PDOException $e) {
```

Comment:
In this `catch`, can we confirm not only a PDOException but check if it's caused by the `locales_source` table missing? If it a PDOException caused by any other problem (such as issue in the query or a failure in the cache or lock), `throw` the exception again to let it bubble up to the calling function.

Reply:
Regarding `throw` I do have one concern: function locale() is run pretty often, potentially many, many times per page call. So this could have severe impact on page performance. Is that acceptable? Or am I missing something?

Reply:
I don't know enough about how try/catch/throw works to understand how it might impact performance. @indigoxela if you've got a chance could you point to info on this?

Reply:
> if you've got a chance could you point to info on this?

I have a question, but not the answer... :wink: That's why I asked.

I'm using try/catch here because it's a "cheap" and lightweight solution - as long as the only `catch` is to immediately return the original string.

But @quicksketch is right, we might hide something else, unrelated to the missing db table, and it might be hard to discover/debug if we silently suppress errors here.

On the other hand function locale() is called very often (by t()). We might throw 50 Exceptions per page call and that must have impact on performance and very likely floods the dblog. So ... what to do?

Reply:
I see.

My feeling is that it is throwing exceptions at all, then performance is not our primary worry. The translation failing is the bigger concern. Or am I missing something?

Reply:
That's a valid point.